WordPress/src/main/java/org/wordpress/android/widgets/AppBarFadingEdgeView.kt

```diff
@@ -0,0 +1,74 @@
+package org.wordpress.android.widgets
+
+import android.R.color
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.drawable.GradientDrawable
+import android.graphics.drawable.GradientDrawable.Orientation.LEFT_RIGHT
+import android.graphics.drawable.GradientDrawable.Orientation.RIGHT_LEFT
+import android.graphics.drawable.StateListDrawable
+import android.util.AttributeSet
+import android.util.StateSet
+import android.view.View
+import androidx.core.content.ContextCompat
+import com.google.android.material.elevation.ElevationOverlayProvider
+import org.wordpress.android.R
+import org.wordpress.android.R.attr
+import org.wordpress.android.util.RtlUtils
+import org.wordpress.android.util.getColorFromAttribute
+
+class AppBarFadingEdgeView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : View(context, attrs, defStyleAttr) {
+    init {
+        iniView()
+    }
+
+    // state_lifted and state_liftable are actually not private, and this waring appears only in code (xml is ok)
+    // is gone on newer versions of gradle
+    @SuppressLint("PrivateResource")
+    private fun iniView() {
+        isDuplicateParentStateEnabled = true
+
+        val elevationOverlayProvider = ElevationOverlayProvider(context)
+        val appbarElevation = resources.getDimension(R.dimen.appbar_elevation)
+        val appBarColor = elevationOverlayProvider.compositeOverlayIfNeeded(
+                context.getColorFromAttribute(R.attr.wpColorAppBar),
+                appbarElevation
+        )
+
+        val fadingEdgeDrawableElevated = GradientDrawable(
+                if (RtlUtils.isRtl(context)) {
+                    LEFT_RIGHT
+                } else {
+                    RIGHT_LEFT
+                },
+                intArrayOf(ContextCompat.getColor(context, color.transparent), appBarColor)
+        )
+
+        val fadingEdgeDrawableFlush = GradientDrawable(
+                if (RtlUtils.isRtl(context)) {
+                    LEFT_RIGHT
+                } else {
+                    RIGHT_LEFT
+                },
+                intArrayOf(
+                        ContextCompat.getColor(context, color.transparent),
+                        context.getColorFromAttribute(attr.wpColorAppBar)
+                )
+        )
+
+        val fadingAgeStateListDrawable = StateListDrawable()
+        fadingAgeStateListDrawable.addState(
+                intArrayOf(
+                        -com.google.android.material.R.attr.state_lifted,
```

Comment:
I'm wondering how this works. Are you intentionally trying to use negative IntResource? or would you expect the value here?

Reply:
Oh man, this is a weird and interesting decision that Android team made. Android is using resource ID integer to indicate the state of the view and since there are are no opposite states (eg. there is no state_disable, or state_not_lifted) they decide to use a negative resource ID integer to indicate opposite state. This kind of makes sense, but I wonder if there are better ways to do it 🤷 

Regardless, it's mentioned in [StateSet](https://developer.android.com/reference/android/util/StateSet) documentation, so I hope I'm not imagining things :)

Reply:
Looking into this makes my head spin. It seems your implementation is correct 👍  but man this is not how you design a standard interface... 😄